_posts/en/newsletters/2019-09-04-newsletter.md

```diff
@@ -0,0 +1,131 @@
+implementations, describes a non-interactive coinjoin proposal, and
+notes a few changes in popular Bitcoin Infrastructure projects.
+
+{% comment %}<!-- include references.md below the fold but above any Jekyll/Liquid variables-->{% endcomment %}
+{% include references.md %}
+
+## Action items
+
+- **Upgrade LN implementations:** one or more issues that are scheduled
+  to be disclosed at the end of this month affect older LN
+  implementations.  If you are using any of the following software
+  versions, upgrading to a more recent version is [strongly
+  recommended:][cve ln]
+
+    - C-Lightning < 0.7.1
+    - LND < 0.7
+    - Eclair <= 0.3
+
+## News
+
+- **SNICKER proposed:** Adam Gibson posted to the Bitcoin-Dev mailing
+  list with a [proposal][snicker] for *Simple Non-Interactive Coinjoin
+  with Keys for Encryption Reused* (SNICKER), a two-step method for
+  allowing wallets to create coinjoins non-interactively.  In the first
+  step---the *proposer* step--Alice's wallet uses the block chain to
+  create a map of addresses to public keys.  The wallet then checks the
+  UTXO set to determine whether any of those addresses have a current
+  balance smaller than the amount Alice controls.  If a satisfactory
+  UTXO is found, Alice creates a proposed coinjoin moving funds from
+  that address's UTXO and one or more of Alice's UTXOs to three outputs:
+
+  1. Coinjoin output to Alice
+
+  2. Coinjoin output to the owner of the selected UTXO (Bob).  Both
+     coinjoin outputs pay the same amount to make them indistinguishable
+     to third parties looking at the block chain
+
+  3. Change output to Alice returning any of her money in excess of the
+     coinjoin amount
+
+  Addresses that already have transaction history can't be reused or the
+  privacy benefits of coinjoin are lost, so Alice generates new unique
+  addresses for her two outputs.  However, Bob has no way in this
+  non-interactive protocol to tell Alice what addresses to use for his
+  output.  Instead, Alice can use Bob's public key and [Elliptic Curve
+  Diffie-Hellman][ECDH] (ECDH) to derive a shared secret that Bob will
+  also be able to derive from Alice's public key.  With the shared
+  secret and Bob's public key, Alice is able to create a new public key
+  that only Bob can sign for.  That new public key is used to create the
+  new address for Bob's coinjoin output.  Nobody besides Alice and Bob
+  can tell the difference between the addresses, ensuring privacy for
+  the coinjoin.
+
+  {% comment %}<!-- Note: the proposal as written assumes the PSBT will
+  be encrypted.  However, I (harding) chatted with Gibson and arubi on
+  IRC about making the PSBTs unencrypted as part of a mechanism to avoid
+  server spam, so the text says "perhaps encrypted" below to leave open
+  both possibilities.-->{% endcomment %}
+
+  With information about the inputs and the outputs, Alice creates a
+  [BIP174][] Partially-Signed Bitcoin Transaction (PSBT) containing the
```

Comment:
I'm going to leave this as-is since I have a busy todo list today and we've used this construction many times before, but tomorrow I'll look into this in more detail and will either get back to you with an explanation or will open a PR to remove the hyphen across the repository.  (Omitting the hyphen feels wrong to me and I have a vague recollection of a rule requiring (or at least suggesting) it, but I need to research it.)

Reply:
Hyphens, thank you. As a recovering liberal hyphenizer before coming across the rule I mentioned, I'm interested in your conclusion. Could imagine a possible exception to the rule for titles or proper nouns.